src/v/cluster/types.h

```diff
@@ -1206,7 +1219,8 @@ struct incremental_topic_updates
           shadow_indexing,
           batch_max_bytes,
           retention_local_target_bytes,
-          retention_local_target_ms);
+          retention_local_target_ms,
+          remote_delete);
```

Comment:
You have to update the ADL encoding too in order not to lose updates.

`IncrementalAlterConfigs` requests are forwarded if they don't land on the controller leader:
https://github.com/redpanda-data/redpanda/blob/b5b4a022904a934836a1aae4976d12268bd49e80/src/v/cluster/topics_frontend.cc#L226-L230

If that's the first request on the connection it will be ADL encoded.

Q: ADL decoding for `incremental_topic_updates` is not forwards compatible? Why is it safe to evolve the encoding?
A: We refuse to update updating topic configuration while in a mixed version state. If the request was always processed on the leader we wouldn't need to update the ADL encoding with actual values.
https://github.com/redpanda-data/redpanda/blob/b5b4a022904a934836a1aae4976d12268bd49e80/src/v/cluster/topics_frontend.cc#L180-L190

Reply:
Added this to the ADL encoding.